### PR TITLE
feat: Add Hello-World Example App

### PR DESCRIPTION
**Pull request by Ehsan-saradar**
This pull request introduces a new feature: a showcase example app demonstrating the process of creating a simple "hello world" application for Ignite. The goal is to provide users with a clear and straightforward guide on how to get started with Ignite by building their first application.

### CODE REVIEW

File: examples/hello-world/go.mod [hunk 1]
```diff
@@ -0,0 +1,201 @@
+module hello-world
+
+go 1.21.1
+
+toolchain go1.21.3
+
+require (
+	github.com/hashicorp/go-plugin v1.5.2
+	github.com/ignite/cli/v28 v28.0.0
```
Comment (jeronimoalbi) [inline examples/hello-world/go.mod:9]: Ignite dependency could be updated to the latest so it includes support for the new `.ign` app extension:

```suggestion
	github.com/ignite/cli/v28 v28.1.1
```

File: examples/hello-world/integration/app_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+				"hello-world",
+			),
+			step.Workdir(app.SourcePath()),
+			step.Stdout(buf),
+		)),
+	))
+	require.Equal("Hello, world!\n", buf.String())
+}
+
+func assertLocalPlugins(t *testing.T, app envtest.App, expectedPlugins []pluginsconfig.Plugin) {
```
Comment (jeronimoalbi) [inline examples/hello-world/integration/app_test.go:57]: The `t.Helper()` calls are helpful when defining helpers to have proper assert messages:

```suggestion
func assertLocalPlugins(t *testing.T, app envtest.App, expectedPlugins []pluginsconfig.Plugin) {
	t.Helper()
```

File: examples/hello-world/integration/app_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	cfg, err := pluginsconfig.ParseDir(app.SourcePath())
+	require.NoError(t, err)
+	require.ElementsMatch(t, expectedPlugins, cfg.Apps, "unexpected local apps")
+}
+
+func assertGlobalPlugins(t *testing.T, app envtest.App, expectedPlugins []pluginsconfig.Plugin) {
```
Comment (jeronimoalbi) [inline examples/hello-world/integration/app_test.go:63]:
```suggestion
func assertGlobalPlugins(t *testing.T, app envtest.App, expectedPlugins []pluginsconfig.Plugin) {
	t.Helper()
```

File: examples/hello-world/main.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	hplugin "github.com/hashicorp/go-plugin"
+
+	"hello-world/cmd"
+
+	"github.com/ignite/cli/v28/ignite/services/plugin"
```
Comment (jeronimoalbi) [inline examples/hello-world/main.go:11]: Import sorting:

```suggestion
	hplugin "github.com/hashicorp/go-plugin"
	"github.com/ignite/cli/v28/ignite/services/plugin"

	"hello-world/cmd"
```

File: examples/hello-world/go.mod (hunk 1 above)
Comment (Ehsan-saradar) [reply examples/hello-world/go.mod]: Fixed

**Comment by toschdev** [comment]
Good work Ehsan!

**Review by julienrbrt — APPROVED** [review]
utACK!

**Pull request merged**